controllers/machinedeployment_controller.go

```diff
@@ -148,6 +154,26 @@ func (r *MachineDeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	return result, err
 }
 
+func patchMachineDeployment(ctx context.Context, patchHelper *patch.Helper, d *clusterv1.MachineDeployment, options ...patch.Option) error {
+	// Always update the readyCondition by summarizing the state of other conditions.
+	conditions.SetSummary(d,
+		conditions.WithConditions(
+			clusterv1.AvailableCondition,
```

Comment:
As it is now does this mean there's no actual semantic difference between available and ready? are we planning to follow up with more conditions here?

Reply:
Yes, I am. Please see the [tracking issue](https://github.com/kubernetes-sigs/cluster-api/issues/3486).